### PR TITLE
ENH: add int64 variant to scotch

### PR DESCRIPTION
**Pull request by olesenm**
- same idea as metis +int64

### CODE REVIEW

File: var/spack/repos/builtin/packages/scotch/package.py [hunk 1]
```diff
@@ -87,9 +89,13 @@ def configure(self):
             '-DCOMMON_RANDOM_FIXED_SEED',
             '-DSCOTCH_DETERMINISTIC',
             '-DSCOTCH_RENAME',
-            '-DIDXSIZE64'
+            '-DIDXSIZE64', # SCOTCH_Idx typedef: indices for addressing
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/scotch/package.py]: Add one more space before the comment symbol to make flake8 happy.

**Comment by olesenm** [comment]
made style change. Added patch for non-threaded, which is not used yet by spack but IMO good to include as safety against future surprises (especially since this is a known scotch issue that hasn't been fixed - project is slowly dormant).

File: var/spack/repos/builtin/packages/scotch/package.py [hunk 2]
```diff
@@ -31,7 +31,7 @@ class Scotch(Package):
        partitioning, graph clustering, and sparse matrix ordering."""
 
     homepage = "http://www.labri.fr/perso/pelegrin/scotch/"
-    url = "http://gforge.inria.fr/frs/download.php/latestfile/298/scotch_6.0.3.tar.gz"
+    url = "http://gforge.inria.fr/frs/download.php/latestfile/298/scotch_6.0.3.tar.gz"  # noqa: E501
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/scotch/package.py]: For the record, we ignore lines containing `https?:` when doing our flake8 line length checks.

Comment (olesenm) [inline var/spack/repos/builtin/packages/scotch/package.py]: good to know.

**Pull request merged**